scheduling/storage_provider_tests.go

```diff
@@ -0,0 +1,69 @@
+package scheduling
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestMemoryStorageProviderSet(t *testing.T) {
+	tests := []struct {
+		key   string
+		value string
+		err   error
+	}{
+		{"key1", "value1", nil},
+		{"", "value1", fmt.Errorf("key cannot be empty")},
+		{"key1", "", fmt.Errorf("value cannot be empty")},
+	}
+
+	var storageProvider MemoryStorageProvider
+
+	for _, test := range tests {
+		err := storageProvider.Set(test.key, test.value)
+		if err != nil && err.Error() != test.err.Error() {
+			t.Errorf("Set(%s, %s): expected error %v, got %v", test.key, test.value, test.err, err)
+		}
+	}
+}
+
+func TestMemoryStorageProviderGet(t *testing.T) {
+	var storageProvider MemoryStorageProvider
+	storageProvider.Set("key1", "value1")
+	storageProvider.Set("key2", "value2")
+	storageProvider.Set("prefix_key3", "value3")
+	storageProvider.Set("prefix_key4", "value4")
+
+	tests := []struct {
```

Comment:
What would help this be more clear, is to add a test name for each case, then use t.Run() in the loop below. Otherwise you have to dig through the logs in order to figure out which one actually failed